mlflow/server/js/src/components/ExperimentView.js

```diff
@@ -31,22 +32,31 @@ class ExperimentView extends Component {
     onSearch: PropTypes.func.isRequired,
     runInfos: PropTypes.arrayOf(RunInfo).isRequired,
     experiment: PropTypes.instanceOf(Experiment).isRequired,
+
+    // List of all parameter keys available in the runs we're viewing
     paramKeyList: PropTypes.arrayOf(String).isRequired,
+    // List of all metric keys available in the runs we're viewing
     metricKeyList: PropTypes.arrayOf(String).isRequired,
-    // List of list of params.
+
+    // List of list of params in all the visible runs
     paramsList: PropTypes.arrayOf(Array).isRequired,
-    // List of list of metrics.
+    // List of list of metrics in all the visible runs
     metricsList: PropTypes.arrayOf(Array).isRequired,
 
-    // Set of paramKeys to include
-    paramKeyFilterSet: PropTypes.instanceOf(Set).isRequired,
-    // Set of metricKeys to include
-    metricKeyFilterSet: PropTypes.instanceOf(Set).isRequired,
+    // Input to the paramKeyFilter field
+    paramKeyFilter: PropTypes.instanceOf(KeyFilter).isRequired,
+    // Input to the paramKeyFilter field
+    metricKeyFilter: PropTypes.instanceOf(KeyFilter).isRequired,
 
-    // The initial searchInput.
+    // The initial searchInput
     searchInput: PropTypes.string.required,
   };
 
+  static defaultProps = {
+    paramKeyFilter: new KeyFilter(),
```

Comment:
Let's remove these default props if the props are declared to be required above. It covers up a kind of quirky bug where `paramKeyFilterSet` was not renamed to `paramKeyFilter` in the `ExperimentViewPage`.

Reply:
Ah, good catch. I probably added it because of that bug...